File: tgrambeta_bot.py
from telegram.ext import Updater, MessageHandler, CommandHandler, Filters, Job
from telegram import KeyboardButton, ParseMode, ReplyKeyboardMarkup
import codecs
import json

import logging
logging.basicConfig(format="\n%(levelname)s: @'%(asctime)s' in '%(name)s':\n> %(message)s", level=logging.INFO)

import os
os.chdir("/home/zlyfer/TelegramBots/TelegramGetTelegramBetaBot/")

updater = Updater(token=open('token', 'r').readline())

tgrambetaapk_file = codecs.open('file_id', 'r')
global tgrambetaapk_file_id
tgrambetaapk_file_id = tgrambetaapk_file.readline()
tgrambetaapk_file.close

global language
language = {}
global langkeyboard
langkeyboard = []

def sys_updateapk(file_id):
    tgrambetaapk_file = codecs.open('file_id', 'w')
    tgrambetaapk_file.write(file_id)
    tgrambetaapk_file.close
    return

def sys_adduser(chat_id):
    users_file = codecs.open('users', 'r+', 'utf-8')
    users_file_content = users_file.readlines()
    exists = False
    for entry in users_file_content:
        if ' %s ' % chat_id in entry:
            exists = True
    if exists == False:
        users_file.write(' %s en-US\n' % chat_id)
    users_file.close
    return

def sys_updatelanguage(chat_id, language_code):
    users_file = codecs.open('users', 'r', 'utf-8')
    users_file_content = users_file.readlines()
    users_file.close
    users_file = codecs.open('users', 'w', 'utf-8')
    for entry in users_file_content:
        if str(chat_id) in entry:
            users_file.write(" %s %s\n" % (chat_id, language_code))
        else:
            users_file.write(entry)
    users_file.close
    return

def sys_getuserlanguage(chat_id):
    users_file = codecs.open('users', 'r', 'utf-8')
    users_file_content = users_file.readlines()
    users_file.close
    for entry in users_file_content:
        if ' %s ' % chat_id in entry:
            return (entry.split()[1])
    return (False)

def job_loadlanguages(bot, job):
    global language
    language = {}
    global langkeyboard
    langkeyboard = []
    for lang in os.listdir('lang'):
        langfile = codecs.open('lang/' + lang, 'r', 'utf-8')
        lang = lang.replace('.json', '')
        langdct = json.loads(langfile.readline())
        language[lang] = langdct
        language[lang]['lang_code'] = lang
        langkeyboard.append([KeyboardButton("%s %s" % (language[lang]['flag'], language[lang]['languagename']))])
        langfile.close
    return

def bot_start(bot, update):

    bot.send_chat_action(chat_id=update.message.chat_id, action='TYPING')
    bot.sendMessage(chat_id=-248828335, parse_mode="HTML", text="<strong>%s/%s/%s/%s</strong>: <i>%s</i>" % (update.message.chat.username, update.message.chat.first_name, update.message.chat_id, sys_getuserlanguage(update.message.chat_id), update.message.text))

    sys_adduser(update.message.chat_id)

    Greeting = ""
    for lang in language:
        Select = language[lang]['flag'] + " " + language[lang]['select'] + "\n"
        Greeting += Select
    update.message.reply_text(parse_mode='HTML', text=Greeting, reply_markup=ReplyKeyboardMarkup(langkeyboard))

    return

def bot_main(bot, update):

    bot.send_chat_action(chat_id=update.message.chat_id, action='TYPING')

    text = update.message.text
    chat_id = update.message.chat_id
    user_lang_code = sys_getuserlanguage(chat_id)
    user_lang = language[user_lang_code]
    username = update.message.chat.username
    first_name = update.message.chat.first_name

    bot.sendMessage(chat_id=-248828335, parse_mode="HTML", text="<strong>%s/%s/%s/%s</strong>: <i>%s</i>" % (username, first_name, chat_id, user_lang_code, text))

    for lang in language:
        if text == language[lang]['flag'] + " " + language[lang]['languagename']:
            sys_updatelanguage(chat_id, language[lang]['lang_code'])
            update.message.reply_text(parse_mode='HTML', text=language[lang]['start'], reply_markup=ReplyKeyboardMarkup([[KeyboardButton("%s" % language[lang]['button'])]]))
            return
        elif text == language[lang]['button']:
            bot.send_document(chat_id=chat_id, document=tgrambetaapk_file_id)
            update.message.reply_text(parse_mode='HTML', text=language[lang]['send'], reply_markup=ReplyKeyboardMarkup([[KeyboardButton("%s" % language[lang]['button'])]]))
            return

    update.message.reply_text(parse_mode='HTML', text=user_lang['unknown'])

    return

def bot_updateapk(bot, update):

    bot.send_chat_action(chat_id=update.message.chat_id, action='TYPING')

    chat_id = update.message.chat_id
    user_lang_code = sys_getuserlanguage(chat_id)
    user_lang = language[user_lang_code]
    username = update.message.chat.username
    first_name = update.message.chat.first_name

    if update.message.chat_id in [175576819]:
        global tgrambetaapk_file_id
        tgrambetaapk_file_id = update.message.document.file_id
        sys_updateapk(tgrambetaapk_file_id)
        update.message.reply_text(parse_mode='HTML', text=user_lang['file'] + "\n<strong>%s</strong>" % tgrambetaapk_file_id)
        bot.sendMessage(chat_id=-248828335, parse_mode='HTML', text="<strong>%s/%s/%s/%s</strong>: <i>FILE_ID:</i>\n<strong>%s</strong>" % (username, first_name, chat_id, user_lang_code, tgrambetaapk_file_id))
    else:
        update.message.reply_text(parse_mode='HTML', text=user_lang['denied'])
        bot.sendMessage(chat_id=-248828335, parse_mode='HTML', text="<strong>%s/%s/%s/%s</strong>: <i>File upload denied!</i>" % (username, first_name, chat_id, user_lang_code))

    return

updater.job_queue.run_repeating(job_loadlanguages, interval=86400, first=0)
updater.dispatcher.add_handler(CommandHandler('start', bot_start))
updater.dispatcher.add_handler(MessageHandler(Filters.text, bot_main))
updater.dispatcher.add_handler(MessageHandler(Filters.document, bot_updateapk))

updater.start_polling()
updater.idle()

File: tgrambeta_bot.js
// jshint esversion: 6
const TelegramBot = require("node-telegram-bot-api");
const https = require("https");
const fs = require("fs");
const token = require("./token.json").token;

const bot = new TelegramBot(token, { polling: true });
const console_chat_id = "-248828335";
const { locals } = require("./locals.json");
var { users } = require("./users.json");

var fileId = require("./fileId.json").fileId;
downloadApk();
setInterval(function() {
  downloadApk();
}, 21600000);

function downloadApk() {}

function addUser(chatId, langCode) {
  users[chatId] = langCode;
  fs.writeFileSync("./users.json", JSON.stringify({ users }));
}

bot.on("text", message => {
  const chatId = message.chat.id;
  const text = message.text;
  const langCode = users[chatId] || message.from.language_code;
  const local = locals[langCode] || locals.en;

  bot.sendMessage(
    console_chat_id,
    `<strong>${message.chat.username}/${message.chat.first_name}/${chatId}/${langCode}</strong>: <i>${text}</i>`,
    {
      parse_mode: "HTML"
    }
  );
  addUser(chatId, langCode);

  // TEMP:
  bot.sendMessage(chatId, `This bot only serves a <strong>outdated version</strong> of Telgram Beta because the download source changed.\nYou will probably still be able to update via the app itself.\nIf you want to download the newest Beta version by yourself head over to: <a href="https://install.appcenter.ms/users/drklo-2kb-ghpo/apps/telegram-beta-2/distribution_groups/all-users-of-telegram-beta-2">App Center</a>\nSorry for the inconvenience, I will try to re-add the automatic download when I have time. Feel free to checkout the <a href="https://github.com/zlyfer/TelegramGetTelegramBetaBot/tree/nodejs">GitHub Repo</a> to contibute.`, { parse_mode: "HTML" });

  switch (text) {
    case "/start":
      bot.sendMessage(
        chatId,
        Object.keys(locals)
          .map(lang => {
            return `${locals[lang].flag} ${locals[lang].select}`;
          })
          .join("\n"),
        {
          parse_mode: "HTML",
          reply_markup: {
            keyboard: Object.keys(locals).map(lang => {
              return [{ text: `${locals[lang].flag} ${locals[lang].languagename}` }];
            })
          }
        }
      );
      break;
    case local.button:
      bot.sendMessage(chatId, local.send, { parse_mode: "HTML" });
      bot.sendDocument(chatId, fileId);
      break;
    default:
      let lang = Object.keys(locals)
        .map(lang => {
          return locals[lang];
        })
        .find(lang => `${lang.flag} ${lang.languagename}` == text);
      if (lang) {
        addUser(chatId, lang.code);
        bot.sendMessage(chatId, locals[lang.code].start, {
          parse_mode: "HTML",
          reply_markup: {
            keyboard: [[{ text: locals[lang.code].button }]]
          }
        });
      } else bot.sendMessage(chatId, local.unknown, { parse_mode: "HTML" });
      break;
  }
});
